platform/mellanox/mlnx-platform-api/sonic_platform_api/fan.py

```
@@ -0,0 +1,231 @@
+                length = fan_pwm.write(str(pwm))
+                print length
+        except (ValueError, IOError):
+            print "failed to open file"
+            status = False
+
+        return status == True
+
+    def set_status_led(self, color):
+        """
+        Set led to expected color
+
+        Returns:
+            bool: True if set success, False if fail. 
+        """
+        if self.is_psu_fan:
+            # PSU fan led status is not able to set
+            return False
+        status = False
+        if color == 'green':
+            try:
+                with open(os.path.join(self.led_path, self.fan_green_led_path), 'w') as fan_led:
+                    fan_led.write(str(LED_ON))
+            except (ValueError, IOError):
+                status = False
+        elif color == 'red':
+            try:
+                with open(os.path.join(self.led_path, self.fan_red_led_path), 'w') as fan_led:
+                    fan_led.write(str(LED_ON))
+            except (ValueError, IOError):
+                status = False
+        elif color == 'off':
+            try:
+                with open(os.path.join(self.led_path, self.fan_green_led_path), 'w') as fan_led:
+                    fan_led.write(str(LED_OFF))
+            except (ValueError, IOError):
+                status = False
+
+            try:
+                with open(os.path.join(self.led_path, self.fan_red_led_path), 'w') as fan_led:
+                    fan_led.write(str(LED_OFF))
+            except (ValueError, IOError):
+                status = False
+        else:
+            pass
+
+        return status == True
+
+    def get_speed_tolerance(self):
+        """
+        Retrieves the speed tolerance of the fan
+
+        Returns:
+            An integer, the percentage of variance from target speed which is
+                 considered tolerable
+        """
+        # The tolerance value not decided yet, discussing with low-level team
+        pass
```

raise NotImplementedError, instead of silent pass

---

this function is under development. will be changed after solution finalized. this is just a place holder.

---

If there will be a fan api user application (fan daemon or smth) it would expect this to return an integer or raise NotImplementedError but right now it will return just None.

If there will be another PR soon than Ok

---

this PR will not be submitted to the community before we find a solution for tolerance. If we can't then will raise NotImplementedError.